helm-charts/charts/kube-starrocks/charts/starrocks/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,47 @@
+{{- if .Values.metrics.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```

Comment:
1. is this a k8s builtin resource type? any external dependencies?
2. how does this differs from the service/pod prometheus annotations?

Reply:
ServiceMonitor is a CRD resource provided by Prometheus. It represents a standard method for metric collection. Compared to the annotation approach, ServiceMonitor allows for more flexible definition of selector and relabeling rules, enabling filtering and modification of metadata for services and endpoints.

Reply:
ServiceMonitor is a CRD resource provided by Prometheus. It represents a standard method for metric collection. Compared to the annotation approach, ServiceMonitor allows for more flexible definition of selector and relabeling rules, enabling filtering and modification of metadata for services and endpoints.

Reply:
see https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-server/servicemonitor.yaml

Reply:
what if the user's env is not deployed with prometheus operator and enables this option in our chart values?

Reply:
`helm upgrade` will fail.

Reply:
is it a proper behavior?

Reply:
In values.yaml, we declared that: 

```
# This configuration is used to integrate with external system Prometheus.
# You can enable the integration by setting the enabled to true.
metrics:
  serviceMonitor:
    # Enable a prometheus ServiceMonitor
    enabled: false
    # Prometheus ServiceMonitor interval
    interval: 15s
```

Reply:
when the helm upgrade/install fails, do you expect the user will know how to fix the failure by installing the prometheus operator?

Reply:
we can add more checks, like this:
```
{{- if and (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") .Values.metrics.serviceMonitor.enabled }}
```

Reply:
ok. update the doc on how to enable prometheus metrics as well since we are redirecting the user to a different approach.